test/replay/networkUtils.test.ts

```diff
@@ -0,0 +1,59 @@
+import { getBodySize, parseContentLengthHeader } from '../../src/js/replay/networkUtils';
+
+describe('networkUtils', () => {
+  describe('parseContentLengthHeader()', () => {
+    it.each([
+      [undefined, undefined],
+      [null, undefined],
+      ['', undefined],
+      ['12', 12],
+      ['abc', undefined],
+    ])('works with %s header value', (headerValue, size) => {
+      expect(parseContentLengthHeader(headerValue)).toBe(size);
+    });
+  });
+
+  describe('getBodySize()', () => {
+    it('works with empty body', () => {
+      expect(getBodySize(undefined)).toBe(undefined);
+      expect(getBodySize(null)).toBe(undefined);
+      expect(getBodySize('')).toBe(undefined);
+    });
+
+    it('works with string body', () => {
+      expect(getBodySize('abcd')).toBe(4);
+      // Emojis are correctly counted as mutliple characters
+      expect(getBodySize('With emoji: 😈')).toBe(16);
+    });
+
+    it('works with URLSearchParams', () => {
+      const params = new URLSearchParams();
+      params.append('name', 'Jane');
+      params.append('age', '42');
+      params.append('emoji', '😈');
```

Comment:
Q: Should we also test with non english characters?

Reply:
Probably not necessary, the main goal is to check that the function doesn't just could string length.